search/search-client-query-transformer/src/main/java/io/camunda/search/clients/DocumentBasedSearchClients.java

```diff
@@ -182,9 +188,58 @@ public List<ProcessDefinitionFlowNodeStatisticsEntity> processDefinitionFlowNode
   @Override
   public SearchQueryResult<ProcessInstanceEntity> searchProcessInstances(
       final ProcessInstanceQuery filter) {
+    if (!filter.filter().incidentErrorHashCodes().isEmpty()) {
+      return searchProcessInstancesByIncidentErrorHash(filter);
+    }
     return getSearchExecutor().search(filter, ProcessInstanceForListViewEntity.class);
   }
 
+  private SearchQueryResult<ProcessInstanceEntity> searchProcessInstancesByIncidentErrorHash(
```

Comment:
🔧 Maybe 1-2 comments would be nice in this method to help people understand what's happening (look up incidents first to get PIs, merge PI keys with original filter etc.)